TACC/Core-Portal
# FP-1647: Use Core-Styles

## Overview: ##

Install [Core-Styles] and use it in various places.

Requires:
- https://github.com/TACC/Core-Styles/pull/23
- https://github.com/TACC/Core-CMS/pull/489

Required by:
- https://github.com/TACC/Core-Portal/pull/653
- https://github.com/TACC/Core-Portal/pull/651

## Related Jira tickets: ##

* [FP-1647](https://jira.tacc.utexas.edu/browse/FP-1647)

## Summary of Changes: ##

- [feat(fp-1647): install core styles](https://github.com/TACC/Core-Portal/commit/188fa1d1b9993fdfb39353f51d431967bd3d027e)
- [feat(fp-1647): use settings from core-styles](https://github.com/TACC/Core-Portal/commit/9c4b072cae5f667729ef2c83aa8d22ab8cf67420)
- [feat(fp-1647): allow shared rulesets via @extend](https://github.com/TACC/Core-Portal/commit/dede2b50ae783b2eaf25027ee4e6d68d417bd7b8)
- [feat(fp-1647): support & use media query features](https://github.com/TACC/Core-Portal/commit/7ec8201a125999f2ad67d35cc3f589342e62bffc)
- [feat(fp-1647): support (local) @import via new path alias](https://github.com/TACC/Core-Portal/commit/efb007348ea4cebb253dc855b6a13d99dfa0ed00)
- [feat(fp-1647): add local media queries](https://github.com/TACC/Core-Portal/commit/72e128d4de6af8fcb85a5d74b08df2c29ec9102e)
- [feat(fp-1647): import o-flex-item-table-wrap](https://github.com/TACC/Core-Portal/pull/639/commits/da002f23cec65a465d3a6fc5ec656beba3ca8f67)
- [feat(fp-1647): compose x-truncate from core-styles](https://github.com/TACC/Core-Portal/pull/639/commits/3b64bf7d8beaef27d5e8400f741859cd7306c1a5)

## Testing Steps & UI Photos: ##

1. <details><summary>Verify <code>.modal-title</code> truncates text with ellipsis.</summary>

    Open a modal, make the title very long (edit code or live markup), and verify title is truncated with an ellipsis.

    ![FP-1647 Test 1](https://user-images.githubusercontent.com/62723358/169911389-0b5a0be5-7228-4344-803f-bd5d89beff7a.png)

    </details>
1. <details><summary>Verify Dashboard and Manage Account still have responsive design.</summary>

    Adjust with the screen width __and__ height. _Some reused media query values now load from [Core-Styles]._

    | A — W 992, H 700) | B — W 991, H 700 | C — W 1200, H 535 | D — W 1200, H 553 |
    | - | - | - | - |
    | ![FP-1647 Test 2A](https://user-images.githubusercontent.com/62723358/169911388-e040a3a3-8d33-4c3e-956e-2def94e6018b.png) | ![FP-1647 Test 2B](https://user-images.githubusercontent.com/62723358/169911387-3b6eeb64-ccb4-4eec-9807-da28accf3773.png) | ![FP-1647 Test 2C](https://user-images.githubusercontent.com/62723358/169911383-627b46cd-3785-46d2-9892-b1ea8e723ae5.png) | ![FP-1647 Test 2D](https://user-images.githubusercontent.com/62723358/169911382-bc6c35aa-cfb3-4cf0-92e0-640894f1dd28.png) |

    </details>
1. <details><summary>Verify UI colors and borders are unchanged.</summary>

    If this failed, colors would be wrong. _The borders and grayscale colors now load from [Core-Styles]._

    | A | B |
    | - | - |
    | ![FP-1647 Test 3A+4A](https://user-images.githubusercontent.com/62723358/169911379-0876993e-fc2c-4b43-9991-da9f5604c8e2.png) | ![FP-1647 Test 3B+4B](https://user-images.githubusercontent.com/62723358/169911370-b59431f3-23fc-4f47-b6bd-13b54c493c1a.png) |

    </details>
1. <details><summary>Verify tables show available content and do not overlap.</summary>

    If this failed, tables would not show content. _The `o-flex-item-table-wrap` fix now loads from [Core-Styles]._

    | A | B |
    | - | - |
    | ![FP-1647 Test 3A+4A](https://user-images.githubusercontent.com/62723358/169911379-0876993e-fc2c-4b43-9991-da9f5604c8e2.png) | ![FP-1647 Test 3B+4B](https://user-images.githubusercontent.com/62723358/169911370-b59431f3-23fc-4f47-b6bd-13b54c493c1a.png) |

    </details>
1. <details><summary>Verify <code>Expand</code>'s <code>.header</code> truncates text with ellipsis.</summary>

    Open a job's View Details modal, edit "System Output" heading text to be very long (edit code or live markup), and verify it is truncated with an ellipsis.

    ![FP-1647 Test 5](https://user-images.githubusercontent.com/62723358/169911372-224fe86c-9685-4372-9fbe-19af307114f0.png)

    </details>
1. <details><summary>Verify <code>ShowMore</code> truncation and reveal works.</summary>

    Open "UI Patterns" section, scroll to "Show More", and verify the text is truncated at 4 lines with an ellipsis.

    Click the "Show More" button link, and verify the text is not truncated.

    | A | B |
    | - | - |
    | ![FP-1647 Test 6A](https://user-images.githubusercontent.com/62723358/169911374-c0fa62c6-b0a6-4e0d-b510-26d16454edee.png) | ![FP-1647 Test 6B](https://user-images.githubusercontent.com/62723358/169911377-22914ba5-29fd-4835-8ec0-1692c9d0f3d1.png) |

    </details>
1. <details><summary>Verify <code>DescriptionList</code> is Truncated.</summary>

    Open "UI Patterns" section, scroll to "DescriptionList", and verify the text is truncated for:
    - "Vertical Layout & Compact Density - Narrow Container"
    - "Horizontal Layout & Compact Density - Narrow Container"

    ![FP-1647 Test 7A+7B](https://user-images.githubusercontent.com/62723358/169911378-3c81cd47-186a-4724-bbeb-acadd2907902.png)

    </details>

## Notes:

<details><summary>I limited code changes for testing Core-Styles. I can do more if you want.</summary>

I intentionally don't use [Core-Styles] everywhere in this PR, so PR is digestible. If you want me to add more relevant changes to **this** PR—e.g. replace all static media queries with custom ones, use extend for even more truncation—just let me know. 🙂

</details>

<details><summary>The diff is very large because of <code>package-lock.json</code>. Seems unavoidable.</summary>

I installed three new dependencies. The `@tacc/core-styles` ([Core-Styles]) dependency is required (or else there is no PR). Each dependency shares a similar large dependency tree, and adds ~8k lines to `package-lock.json`, but together they add less than any one adds individually.

</details>

[Core-Styles]: https://github.com/TACC/Core-Styles
